port to python3

sourabhaa opened the pull request.

ayushnawal committed:
update sugargame to v1.3

ayushnawal committed:
Port to Python 3

ayushnawal committed:
changes made as requested

sourabhaa committed:
fix: fixed sound icon

sourabhaa committed:
activity works but there's issue with pygame initialisation

chimosky commented on activity.py:
```diff
@@ -24,8 +24,8 @@ def __init__(self, handle):
         self._pygamecanvas = sugargame.canvas.PygameCanvas(self,
-            main= self.actividad.make,
-            modules= [pygame, pygame.display, pygame.mixer])
+            main= self.actividad.make(),
```
The function that's usually passed to `main` isn't called at the time it's passed, could you also fix the spaces after the assignment?

sourabhaa replied:
> The function that's usually passed to main isn't called at the time it's passed,

I didn't completely understand, could you please give more info on this?

chimosky replied:
You've called `make()` on line 27 while passing it to `main` and usually the functions that get passed to main are referenced and not called when they're being passed.

sourabhaa replied:
@chimosky @srevinsaju, I'm not able to fix this. Could you please help with it?

chimosky replied:
@quozl can you please take a look at this? I haven't figured out why the game canvas doesn't render like it should.

quozl replied:
Okay, looking into it.

chimosky commented:
Tested d88a680, works as expected.

It's a bit difficult to play the game without any instructions so one thing you can add after this has been merge is a simple instruction on the game play at the start of the game.

sourabhaa committed:
refactor: fixed spacing after the assignmet

quozl committed:
Allow execution of files

Two Python files did not have executable permissions.

Shouldn't cause any symptom, but was irritating.

Signed-off-by: James Cameron <quozl@laptop.org>

quozl committed:
Adopt conventions

- camel case the game class name,

- rename source file to match lowercase class name,

- rename instance variable to match Sugargame test activity

Signed-off-by: James Cameron <quozl@laptop.org>

quozl committed:
Remove show_all in toolbar

Sugargame test activity does not do this.

Was also found in other activities, such as Bridge and Jumble, it
changes the order of creation of the X11 windows.

The earlier port of thie activity to Sugargame 1.3 has not properly
followed the new API and test activity.

Signed-off-by: James Cameron <quozl@laptop.org>

quozl committed:
Fix path to score file

EPERM is reported when the activity is run under Sugar when the source
code is in /usr/share/sugar/activities

quozl committed:
Add missing get_preview method

quozl committed:
Show the sound button

quozl committed:
Port to Sugargame 1.3 API

- name the run method the same as the example program,

- use a running flag rather than a crashed flag; crashed could be
  misunderstood as referring to the cars, and running is how the
  example program is phrased,

- change initialisation of the display,

- properly exit the event loop if the stop button is pressed,

- handle all events in the message queue in one place.

Tested on Ubuntu 20.04 with Sugar.  Mostly works.

Signed-off-by: James Cameron <quozl@laptop.org>

quozl commented:
Thanks for asking, @chimosky, it was an interesting problem.  Please see the patches I've pushed to @sourabhaa's branch.  The root cause was that the activity had not been ported to the changed API of Sugargame 1.3.  Anyone else could have fixed it by looking at those changes to Sugargame and implementing them in the activity.

I ran out of time.  The activity still has a few minor problems, which I won't mention yet; please both test and see what you think.  At least now it works.

sourabhaa commented:
Thanks @quozl , tested [85e2019](https://github.com/sugarlabs/2-cars-activity/pull/10/commits/85e201918e2d7628c7403e6f3bb11d79fff5b643),  activity is working fine now :)

chimosky commented:
> I ran out of time. The activity still has a few minor problems, which I won't mention yet; please both test and see what you think. At least now it works.

Thanks for your changes, I've learnt a couple things with regards to pygame activities. Added a commit.

Firstly, there's this little lag time between when the activity starts and when the welcome screen is displayed and that's why I'd moved the check for that but it doesn't seem to solve the problem.

The other issue is that the game isn't centered, it's more to the right and you have `width` and `height` created but not used, I'll leave @sourabhaa to fix those.

quozl commented:
In 81f2088b I understand the intent behind your move of the welcome screen, but it is unlikely to change the delay you saw.  The event checking steps do not pause execution.  Only the clock does that.

In 81f2088b the removal of 275 and 276 seems very wrong; doesn't that mean that every key will act as left and right keys instead of those keys?

For the start delay, does it occur for you using the Sugargame TestActivity?

chimosky commented:
> In https://github.com/sugarlabs/2-cars-activity/commit/81f2088ba959b53a3a1508d8d4d8beee3805cf81 I understand the intent behind your move of the welcome screen, but it is unlikely to change the delay you saw. The event checking steps do not pause execution. Only the clock does that.

Yeah the event checking steps don't pause execution, haven't investigated to see how to fix this.

> In https://github.com/sugarlabs/2-cars-activity/commit/81f2088ba959b53a3a1508d8d4d8beee3805cf81 the removal of 275 and 276 seems very wrong; doesn't that mean that every key will act as left and right keys instead of those keys?

No it doesn't mean that every key will act like left and right keys and not themselves it means that if the value of those keys were to ever change you don't have to bother about that.
The pygame documentation also suggests this:
> Portability note: The integers for key constants differ between pygame 1 and 2. Always use key constants (K_a) rather than integers directly (97) so that your key handling code works well on both pygame 1 and pygame 2.


>For the start delay, does it occur for you using the Sugargame TestActivity?

No it doesn't happen when I run the Sugargame TestActivity.

quozl commented:
I've tested 81f2088 now, and every key now acts like left and right keys.  The code does not test `event.key` at all, so your reference to key constants is odd.

chimosky commented:
> I've tested [81f2088](https://github.com/sugarlabs/2-cars-activity/commit/81f2088ba959b53a3a1508d8d4d8beee3805cf81) now, and every key now acts like left and right keys. The code does not test `event.key` at all, so your reference to key constants is odd.

Yes, tested and the `event.key` check is needed.  

It does seem odd that the key constant check works for every key press event.

@sourabhaa do you intend on completing this PR?

sourabhaa commented:
Yes, @chimosky I'll work on this. I'm working on the PR in Bridge activity, I'll look into this soon.

Dimi20cen commented:
Should we close this PR?
